vmdb/app/models/mixins/naming_sequence_mixin.rb

```diff
@@ -0,0 +1,19 @@
+module NamingSequenceMixin
+  extend ActiveSupport::Concern
+
+  NAMING = 'naming'
+
+  included do
+    # these are possible naming sequences because there is no constraint to source column
+    has_many :naming_sequences, :as => :resource, :dependent => :destroy, :class_name => "CustomAttribute"
+  end
+
+  def next_naming_sequence(name, source)
+    lock do
+      record = CustomAttribute.where(:name => name, :source => source).first_or_create
```

Comment:
@Fryguy I temporarily changed to CustomAttribute for testing but forgot to change back. Good catch. Thanks.